janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryUtil.java

```diff
@@ -224,6 +223,7 @@ public static boolean isEmpty(Condition<?> condition) {
                 }
             } else if (predicate instanceof OrJanusPredicate) {
                 final List<Object> values = (List<Object>) (value);
+                // FIXME: this might generate a non QNF-compliant form, e.g. nested = PredicateCondition OR (PredicateCondition AND PredicateCondition)
```

Comment:
What can it lead to? Do you think we should open an issue to know about this problem?

Reply:
Sure, I just created https://github.com/JanusGraph/janusgraph/issues/2775.